include/prism/util/pm_string.h

```diff
@@ -35,18 +35,18 @@ typedef struct {
     size_t length;
 
     /** The type of the string. This field determines how the string should be freed. */
-    enum {
+    enum pm_string_type {
```

Comment:
This anonymous enum made broke the docs, but also gave worse in-editor feedback.

Before:

<img width="387" alt="Screenshot 2024-08-30 at 2 48 58 PM" src="https://github.com/user-attachments/assets/76e2c8e5-952d-4060-b25a-2efcf8a92043">

After:

<img width="864" alt="image" src="https://github.com/user-attachments/assets/25ec0fd1-da8e-487b-9814-aef8c5e141fc">